### ISSUE TITLE
Missing IPv6 support

### ISSUE THREAD

**Issue by walter-ebner** [jborean93/pypsexec#17]
When I use a hostname to connect which resolves to an IPv6 address I get the following error:

```
  File "test-pypsexec.py", line 24, in <module>
    c.connect()
  File "/var/lib/rundeck/.shiv/pypsexec_9de8d3fe-ca00-48cc-bfb3-b7bdf840e72b/site-packages/pypsexec/client.py", line 60, in connect
    self.connection.connect(timeout=timeout)
  File "/var/lib/rundeck/.shiv/pypsexec_9de8d3fe-ca00-48cc-bfb3-b7bdf840e72b/site-packages/smbprotocol/connection.py", line 835, in connect
    self.transport.connect(timeout=timeout)
  File "/var/lib/rundeck/.shiv/pypsexec_9de8d3fe-ca00-48cc-bfb3-b7bdf840e72b/site-packages/smbprotocol/transport.py", line 61, in connect
    self._sock.connect((self.server, self.port))
socket.gaierror: [Errno -2] Name or service not known
```
When I use the IPv6 address directly I get the following error:

 ```
 File "test-pypsexec.py", line 24, in <module>
    c.connect()
  File "/var/lib/rundeck/.shiv/pypsexec_9de8d3fe-ca00-48cc-bfb3-b7bdf840e72b/site-packages/pypsexec/client.py", line 60, in connect
    self.connection.connect(timeout=timeout)
  File "/var/lib/rundeck/.shiv/pypsexec_9de8d3fe-ca00-48cc-bfb3-b7bdf840e72b/site-packages/smbprotocol/connection.py", line 835, in connect
    self.transport.connect(timeout=timeout)
  File "/var/lib/rundeck/.shiv/pypsexec_9de8d3fe-ca00-48cc-bfb3-b7bdf840e72b/site-packages/smbprotocol/transport.py", line 61, in connect
    self._sock.connect((self.server, self.port))
socket.gaierror: [Errno -9] Address family for hostname not supported
```
Then I tried the UNC Notation for IPv6 addresses like it is required for SMB connections as in UNC paths there is no colon allowed: 
2001-9d1-3a1-1ee-0-2034-d41f-5d28.ipv6-literal.net
This also doesn't work: socket.gaierror: [Errno -2] Name or service not known

I am looking for a replacement for psexec to reach a thousand servers via IPv6.

Regards,
Walter

**Comment by jborean93** [comment]
This is probably more a problem with [smbprotocol](https://github.com/jborean93/smbprotocol) and not pypsexec as I just pass the connection info to that library there. The issue is because I am [creating the socket](https://github.com/jborean93/smbprotocol/blob/master/smbprotocol/transport.py#L55) with `AF_INET` which is IPv4 or a DNS hostname. I'll need to update the code there to support IPv6 as well, thanks for letting me know about this issue.

_Just a side note, I'm going on holidays for 4 weeks so I won't get to this for some time but I will do it eventually unless you would like to fix it._

**Comment by jborean93** [comment]
I've opened a PR on smbprotocol that should solve this issue https://github.com/jborean93/smbprotocol/pull/13. It would be great if you were able to test it out and verify whether that solves the issue for you.

**Comment by jborean93** [comment]
I've just released `v1.0.0` of smbprotocol which includes this fix.

**Comment by walter-ebner** [comment]
I can confirm that it is working properly using IPv6 addresses or DNS names that are resolving to IPv6 addresses. Link Local IPv6 addresses I didn't try.
